3-networks-hub-and-spoke/README.md

```diff
@@ -0,0 +1,315 @@
+# 3-networks-hub-and-spoke
+
+This repo is part of a multi-part guide that shows how to configure and deploy
+the example.com reference architecture described in
+[Google Cloud security foundations guide](https://services.google.com/fh/files/misc/google-cloud-security-foundations-guide.pdf)
+(PDF). The following table lists the parts of the guide.
+
+<table>
+<tbody>
+<tr>
+<td><a href="../0-bootstrap">0-bootstrap</a></td>
+<td>Bootstraps a Google Cloud organization, creating all the required resources
+and permissions to start using the Cloud Foundation Toolkit (CFT). This
+step also configures a CI/CD pipeline for foundations code in subsequent
+stages.</td>
+</tr>
+<tr>
+<td><a href="../1-org">1-org</a></td>
+<td>Sets up top level shared folders, monitoring and networking projects, and
+organization-level logging, and sets baseline security settings through
+organizational policy.</td>
+</tr>
+<tr>
+<td><a href="../2-environments"><span style="white-space: nowrap;">2-environments</span></a></td>
+<td>Sets up development, non-production, and production environments within the
+Google Cloud organization that you've created.</td>
+</tr>
+<tr>
+<td><a href="../3-networks-dual-svpc">3-networks-dual-svpc</a></td>
+<td>Sets up base and restricted shared VPCs with default DNS, NAT (optional),
+Private Service networking, VPC service controls, on-premises Dedicated
+Interconnect, and baseline firewall rules for each environment. It also sets
+up the global DNS hub.</td>
+</tr>
+<tr>
+<td><a>3-networks-hub-and-spoke (this file)</a></td>
+<td>Sets up base and restricted shared VPCs with all the default configuration
+found on step 3-networks-dual-svpc, but here the architecture will be based on the
+Hub and Spoke network model. It also sets up the global DNS hub</td>
+</tr>
+</tr>
+<tr>
+<td><a href="../4-projects">4-projects</a></td>
+<td>Sets up a folder structure, projects, and application infrastructure pipeline for applications,
+ which are connected as service projects to the shared VPC created in the previous stage.</td>
+</tr>
+<tr>
+<td><a href="../5-app-infra">5-app-infra</a></td>
+<td>Deploy a simple <a href="https://cloud.google.com/compute/">Compute Engine</a> instance in one of the business unit projects using the infra pipeline set up in 4-projects.</td>
+</tr>
+</tbody>
+</table>
+
+For an overview of the architecture and the parts, see the
+[terraform-example-foundation README](https://github.com/terraform-google-modules/terraform-example-foundation).
+
+## Purpose
+
+The purpose of this step is to:
+
+- Set up the global [DNS Hub](https://cloud.google.com/blog/products/networking/cloud-forwarding-peering-and-zones).
+- Set up base and restricted shared VPCs with default DNS, NAT (optional), Private Service networking, VPC service controls, on-premises Dedicated or Partner Interconnect, and baseline firewall rules for each environment.
```

Comment:
Lets update this to reflect hub and spoke model i.e base and restricted hubs and corresponding spokes.